### PR TITLE
Test: prove the sim runner publishes launch acceptance

### PR DESCRIPTION
**Pull request by ChaoWao**
## Summary

Adds the missing regression barrier for launch-acceptance publication on sim.

**This PR started as the strict-symbol-loading refactor and has been narrowed to
the test.** #1587 merged as `2a650f2d` while this was open and landed the whole
refactor independently — all four symbols strict, `load_optional_symbol`
deleted, the sim `set_task_accepted_state_ctx` export, the dead nullptr guards
removed, and the header docs updated. I verified main against my diff
line-by-line and it is equivalent throughout, so I dropped my `src/` changes
wholesale on rebase rather than churn a mechanism that had already shipped.

The one difference in approach: for `get_pipeline_contract` — absent from a5's
runtime makers — I had put a **weak depth-1 default in both platform
`c_api_shared.cpp`**, matching `prewarm_config_impl`; main instead added
**strong definitions to a5's two `runtime_maker.cpp`**. Both give a5 the
depth-1 contract, main's is explicit at the cost of the literal appearing in
two more places, and I checked a5's `host_build_graph` definition sits inside
`extern "C" {` so its linkage is right. Main's version stands.

## What is left, and why it is worth landing

Nothing asserted the accepted word ever **arrives**. The two endpoint scene
tests that read it (`worker_async_endpoint`, `worker_async_fifo`) assert it is
still `0` before activation — the negative direction — and both are
a2a3-onboard-only, so the sim side of the binding had no coverage at all.

That gap hid a real defect until #1587. The sim c_api exported no
`set_task_accepted_state_ctx`, so ChipWorker's then-optional load produced
`nullptr`, both bind sites were skipped, and `SimDeviceRunnerBase::publish_task_accepted`
found a null target. A sim child therefore never published acceptance, and the
run-level fence — `LocalMailboxEndpoint::read_task_accepted` → `on_accept` →
`Orchestrator::mark_task_accepted` → `decrement_run_accepts` — advanced only
once the run reached a terminal phase, because `acceptance_ready()` was
satisfied by its other disjunct. The launch fence silently degraded into a
completion fence on every sim run.

`worker.py` passes the accepted-word address on every platform (single-frame
`run_from_blob` and two-frame `_launch_native_run`, neither platform-gated), and
the sim runner does reach `publish_task_accepted()` at its launch boundary — so
this was never "sim doesn't use the accepted word", only an unexported symbol.

The test dispatches one ChipTask and asserts the word is set in whichever
mailbox frame carried it. That holds on both endpoint shapes because the parent
clears the word only when it publishes the next task into that frame, so it
survives completion.

**Verified to actually catch the regression:** against the pre-#1587 sim c_api
it fails with

```
E  AssertionError: the chip worker never published launch acceptance: [0, 0, 0]
E  assert 1 in [0, 0, 0]
```

## Testing

At the rebased base (`2a650f2d`):

- a2a3sim — this test: passed
- a2a3 onboard via `task-submit` (`task_20260803_010730_25475769728`, 1 device) — this test: **passed on real silicon**
- pre-commit on the changed file: all hooks passed

From the pre-rebase run of the full refactor (superseded by #1587, but the test
was present throughout): cpput 74/74 · pyut 1034 passed / 9 skipped · a2a3sim
sweep 63 PASS 0 FAIL · a5sim sweep 47 PASS 0 FAIL · a2a3 onboard sweep 62 PASS
0 FAIL before the run hit the 600 s `--pto-session-timeout` budget in its last
group on this shared box.

### CODE REVIEW

File: tests/st/a2a3/tensormap_and_ringbuffer/test_l3_launch_acceptance.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+    CASES = [
+        {
+            "name": "default",
+            "platforms": ["a2a3sim", "a2a3"],
+            "config": {"device_count": 1, "num_sub_workers": 0, "aicpu_thread_num": 4},
+            "params": {},
+        },
+    ]
```
[the remark this answers is not in the document]
Comment (ChaoWao) [reply tests/st/a2a3/tensormap_and_ringbuffer/test_l3_launch_acceptance.py:91]: Skipping this one.

`RUF` is not in this repo's ruff `select` list — `pyproject.toml:37-44` selects `PL`, `I`, `E`, `W`, `F`, `UP` only — so RUF012 is not a rule enforced here. The repo's own `ruff check` passes on this file in pre-commit, locally and in CI.

It would also break with local style: all 83 scene tests under `tests/st/` declare `CALLABLE = {` / `CASES = [` bare, and none uses `ClassVar`. Annotating only this file would make it the outlier, which `.claude/rules/discipline.md` §2 ("match local style when editing") steers away from. If the project wants the annotation it belongs in a sweep plus a ruff `select` entry to hold the line, not in one new test.

**Comment by ChaoWao** [comment]
Narrowed after #1587 landed.

#1587 merged as `2a650f2d` while this was open and independently landed the strict-symbol-loading refactor this PR opened with — all four symbols on `load_symbol`, `load_optional_symbol` deleted, the sim `set_task_accepted_state_ctx` export, the dead nullptr guards dropped, and the header docs updated. I diffed main against my version and it is equivalent throughout, so I took main's `src/` wholesale on rebase instead of churning a mechanism that already shipped.

One approach difference, resolved in main's favour: for `get_pipeline_contract` I had a weak depth-1 default in both platform `c_api_shared.cpp` (mirroring `prewarm_config_impl`); main added strong definitions to a5's two `runtime_maker.cpp`. Same outcome for a5, and I checked a5's `host_build_graph` definition is inside `extern "C" {`, so its linkage is correct.

What is left is the test, which main has no equivalent of — the two endpoint tests that read the acceptance word assert it is still `0` before activation and are a2a3-onboard-only, so the sim binding had no coverage. It is verified to fail against the pre-#1587 sim c_api. Body updated with the full rationale.
